### PR TITLE
Fix/ca 27710/fix line items already fulfilled error message

### PR DESCRIPTION
**Pull request by clericuzzi**
fixes fulfillment order based fulfillments tracking information

### CODE REVIEW

**Comment by rjdavis3** [comment]
Task linked: [CA-27710 Shopify Order Fulfillment Line Items are required error when line item is already fulfilled](https://app.clickup.com/t/2241076/CA-27710)

File: src/test/java/com/shopify/ShopifySdkTest.java [hunk 1]
```diff
@@ -447,6 +449,77 @@ public void givenSomeShopifyFulfillmentCreationRequestWhenCreatingShopifyFulfill
...
+				giveResponse(expectedResponseBodyString, MediaType.APPLICATION_JSON).withStatus(expectedStatusCode)
+						.withHeader(ShopifySdk.DEPRECATED_REASON_HEADER, "Call to be removed from API.")
+						.withHeader("Location", new StringBuilder().append("https://test.myshopify.com/admin")
+								.append(expectedPath).toString()));
+
+		final ShopifyFulfillmentCreationRequest request = ShopifyFulfillmentCreationRequest.newBuilder()
+				.withOrderId("1234").withTrackingCompany("USPS").withTrackingNumber("12341234").withNotifyCustomer(true)
+				.withLineItems(Arrays.asList(lineItem)).withLocationId("1").withTrackingUrls(Arrays.asList()).build();
+
+		request.getRequest().setTrackingUrl("tracking_url");
```
Comment (ryankazokas) [inline src/test/java/com/shopify/ShopifySdkTest.java:515]: i don't see a test that tests that branch of code. Can you add one small unit test here before merging that checks that we are setting the first tracking url if there are multiple

**Review by ryankazokas — CHANGES_REQUESTED** [review]

File: src/test/java/com/shopify/ShopifySdkTest.java (hunk 1 above)
Comment (clericuzzi) [reply src/test/java/com/shopify/ShopifySdkTest.java:515]: added LegacyToFulfillmentOrderMappingTest to go over the trackingUrl checks

**Review by ryankazokas — APPROVED** [review]
Much better on the tests! Thank you! Feel free to merge when ready

**Pull request merged**